package.json

```diff
@@ -1,17 +1,20 @@
 {
   "repository": "https://github.com/kiltprotocol/sdk-js",
+  "name": "root-workspace",
   "private": true,
   "workspaces": [
     "packages/*",
     "docs/*"
   ],
   "license": "BSD-4-Clause",
   "scripts": {
-    "check": "tsc -p tsconfig.json --noEmit",
-    "build": "yarn workspaces foreach -p -t --exclude '{root-workspace}' run build",
+    "check": "tsc -b tsconfig.json",
+    "build": "yarn clean && yarn workspaces foreach -pt --exclude 'root-workspace' run build",
+    "build:dev": "yarn workspaces foreach -pt --exclude 'root-workspace' run build:cjs",
+    "build:watch": "yarn build:dev && tsc -b tsconfig.json --watch",
```

Comment:
What's the benefit of this?

Reply:
It complains as soon as you break something and is more efficient as running `yarn check` or `yarn build:dev` repeatedly. Also I would have to double check but potentially you need to have it running if you are using jest in watch mode.

Reply:
just checked, you would want to have this running as well if jest is running in watch mode to make sure packages are being rebuild on changes. Otherwise imports from another package may use out of date versions.

Reply:
So this is supposed to be run before and during tests? And `yarn build:dev` alone then when?

Reply:
Again, this is for watch mode. As in, you keep jest running and it re-executes tests whenever you change a file. So far we didn't have a solution for that. You can also just run `yarn build:dev && yarn test` whenever you want to test of course. It's a question of preference.

As for the naming, I was considering making `yarn build:dev` the new `yarn build` and introducing a `yarn build:production` instead for release builds. Do you think that's better? I decided to go the other way because there is no harm in running the production build in a development scenario (it's just slower) but it would be bad to release a development build because the esm files are missing.

Reply:
This makes sense. Rather use the build for development and something else for production.